src/vm/types/mod.rs

```diff
@@ -1139,6 +1139,35 @@ impl TupleData {
             CheckErrors::NoSuchTupleField(name.to_string(), self.type_signature.clone()).into()
         })
     }
+
+    pub fn deep_merge(base: TupleData, updates: TupleData) -> Result<TupleData> {
+        let mut base = base;
+        for (name, value) in updates.data_map.into_iter() {
+            let (_, existing_value) =
+                base.data_map
+                    .remove_entry(&name)
+                    .ok_or(CheckErrors::NoSuchTupleField(
```

Comment:
This wasn't clear from the documentation:  `(set a b)` only works if all keys in `b` exist in `a`, right?  For example, this should fail to type-check: `(set { a: 1 } { b: 2 })`?

Reply:
A `merge` function should return the merged properties, resulting in `{ a: 1, b: 2}`.